app/controllers/hiring_staff/base_controller.rb

```diff
@@ -28,4 +32,23 @@ def current_user
 
     @current_user ||= User.find_or_create_by(oid: current_session_id)
   end
+
+  def check_user_last_activity_at
+    return redirect_to dsi_logout_url if current_user&.last_activity_at.blank?
+
+    if Time.zone.now > (current_user.last_activity_at + TIMEOUT_PERIOD)
+      session[:signing_out_for_inactivity] = true
+      redirect_to dsi_logout_url
```

Comment:
I think perhaps this should be `return redirect_to`, in order to make the action `check_user_last_activity_at` closed and thus avoid the possibility of too many redirects.